### PR TITLE
add API calls to get category and videos

### PR DESCRIPTION
**Pull request by hsnice16**
### This PR will add code for the API calls to get all the categories and videos

In this PR, **+895 additions -111 deletions** lines of code
- some files contain only 5 to 6 lines of code
- if you don't prefer you can avoid reviewing Backend related files

### Please review this PR, and give your feedback

---
#### [Live URL](https://dekho-video-53fhe6dao-hsnice16.vercel.app/)

image

### CODE REVIEW

File: src/reducer/constants/shared-action-type.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export const ACTION_TYPE_LOADING = "LOADING";
+export const ACTION_TYPE_SUCCESS = "SUCCESS";
+export const ACTION_TYPE_ERROR = "ERROR";
```
Comment (vrinda-mahajan) [inline src/reducer/constants/shared-action-type.js:3]: Liked that you made constants for the dispatch type.

**Review by vrinda-mahajan — APPROVED** [review]
Appreciation:
Your code is clean and readable.
The folder structure is also awesome.
Liked that you have used constants for dispatch values.

Just a suggestion on UI, on clicking somewhere else on the screen the settings component should go away.
Screenshot (95)

**Comment by hsnice16** [comment]
> Appreciation: Your code is clean and readable. The folder structure is also awesome. Liked that you have used constants for dispatch values.
> 
> Just a suggestion on UI, on clicking somewhere else on the screen the settings component should go away. Screenshot (95)

Thanks for making the suggestion completely clear by attaching the SS. Yeah, I will do the changes.

File: src/pages/home/Home.jsx [hunk 2]
```diff
@@ -1,32 +1,53 @@
...
+  const { status, data } = categories;
+  const { state: videos } = useAsync(API_TO_GET_ALL_VIDEOS);
+  const filteredVideos =
+    videos.status === "success"
+      ? videos.data.filter((video) =>
+          selectedCategory === "All"
+            ? video
+            : video.categoryName === selectedCategory
+        )
+      : [];
```
Comment (mikesch803) [inline src/pages/home/Home.jsx]: appreciation : liked the way you trim the fat

**Review by mikesch803 — APPROVED** [review]
your code really great, you can merge this

File: src/custom-hooks/useCategory.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      dispatch,
+      setSelectedCategory,
+      getCategoryFilteredData,
+    }
+ */
+export const useCategory = () => {
+  const [selectedCategory, setSelectedCategory] = useState("All");
+  const { state, dispatch } = useAsync(API_TO_GET_ALL_CATEGORIES);
+
+  /**
```
Comment (jalajcodes) [inline src/custom-hooks/useCategory.js:22]: Apperciation:  its nice that you have added jsdoc comments.

Comment (jalajcodes) [inline src/custom-hooks/useCategory.js:22]: Apperciation: its nice that you have added jsdoc comments.

**Review by jalajcodes — APPROVED** [review]
Well written code. Good work Himanshu!

**Pull request merged**